src/config/mongoose_config_validator_toml.erl

```diff
@@ -429,8 +1551,18 @@ validate_non_negative_integer(Value) when is_integer(Value), Value >= 0 -> ok.
 validate_non_negative_integer_or_infinity(Value) when is_integer(Value), Value >= 0 -> ok;
 validate_non_negative_integer_or_infinity(infinity) -> ok.
 
+validate_positive_integer_or_infinity(Value) when is_integer(Value), Value > 0 -> ok;
+validate_positive_integer_or_infinity(infinity) -> ok.
+
 validate_enum(Value, Values) ->
-    true = lists:member(Value, Values).
+    case lists:member(Value, Values) of
```

Comment:
I'm seeing these error wrappers popping at some places - is that random (as a test fails and you see a message like 'badmatch', you change this) or is there a motivation for doing it here and not for the other validator functions? I think we should keep it consistent. Any motivation might be good, could you just share it?

Reply:
It's the only (ok, there is validate_backend) function that takes an extra argument, which is not visible in the error.

Reply:
That's a reason, so let's keep it :-)